### PR TITLE
ci: update build and add worflow for maven central publishing

### LINKED ISSUE #3: Prepare for maven publish
Changes to publish to maven central and also insee maven internal repo

### PR DESCRIPTION
**Pull request by FBibonne**
- feat: allow publishing in custom repo with command line argument
- ci: add release workflow (that publishes on maven central)
- ci: add test workflow
- build: update gradle wrapper
- fix gpg variable names which were confusing in build.gradle
- chore: bump version to release 1.0.0

### CODE REVIEW

File: model/build.gradle.kts [hunk 1]
```diff
@@ -109,27 +111,44 @@ tasks {
...
-            val releaseRepo: URI = URI.create("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+            val snapshotRepo2: URI = URI.create("https://s01.oss.sonatype.org/content/repositories/snapshots/")
+            val releaseRepo2: URI = URI.create("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+            val mavenRepo: URI = if (isSnapshot) snapshotRepo2 else releaseRepo2
+
+            val commandLineRepoUrl = System.getProperty("repoUrl")
 
             name = "OSSRH"
-            url = if (isSnapshot) snapshotRepo else releaseRepo
+            url = if (commandLineRepoUrl != null) URI.create(commandLineRepoUrl) else mavenRepo
```
Comment (nsenave) [inline model/build.gradle.kts:133]: simplified this section: by default `gradle publish` publishes on maven central, and a custom repo can be passed `gradle publish -DrepoUrl="..."`

File: model/build.gradle.kts [hunk 2]
```diff
@@ -26,7 +26,8 @@ java {
 }
 
 group = "fr.insee.ddi"
-version = "0.1.0-SNAPSHOT"
+version = "1.0.0"
+val nameForArtifactAndJar by extra("ddi-lifecycle")
```
Comment (nsenave) [inline model/build.gradle.kts:30]: I don't know kotlin/gradle enough: I'm curious about this "by extra"

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
```
Comment (nsenave) [inline gradle/wrapper/gradle-wrapper.properties:3]: in the meantime version 8.6 has been released

File: model/build.gradle.kts [hunk 4]
```diff
@@ -166,10 +185,10 @@ publishing {
...
 signing {
     // Environment variables for ascii-armored keys (to be used in CI)
-    val publicGPGKey = System.getenv("GPG_PUBLIC_KEY")
-    val privateGPGKey = System.getenv("GPG_PRIVATE_KEY")
-    if (publicGPGKey != null && privateGPGKey != null) {
-        useInMemoryPgpKeys(publicGPGKey, privateGPGKey)
+    val privateGPGKey = System.getenv("GPG_SIGNING_KEY")
+    val gpgPassword = System.getenv("GPG_PASSWORD")
+    if (privateGPGKey != null && gpgPassword != null) {
+        useInMemoryPgpKeys(privateGPGKey, gpgPassword)
```
Comment (nsenave) [inline model/build.gradle.kts:191]: I didn't name variables correctly which was confusing

**Review by nsenave — APPROVED** [review]
Added a publish github action workflow, I'll merge the branch to trigger it, we still can change anything in a next pr if we want

**Comment by nsenave** [comment]
Note: for now the CI only authorize publishing release versions (no snapshots)

I'll add a snapshot workflow if necessary

**Comment by FBibonne** [comment]
Thank you very much @nsenave

File: model/build.gradle.kts (hunk 2 above)
Comment (FBibonne) [reply model/build.gradle.kts:30]: I picked it from this example : https://docs.gradle.org/current/userguide/writing_build_scripts.html#sec:extra_properties

**Pull request merged**